demisto_sdk/commands/lint/lint_manager.py

```diff
@@ -109,30 +114,56 @@ def __init__(
             base_branch=self._prev_ver,
         )
 
-        self._id_set_path = id_set_path
         if check_dependent_api_module:
-            print(
-                "Checking for packages dependent on the modified API module", end="... "
-            )
-            dependent_on_api_module = get_api_module_dependencies(
-                self._pkgs, self._id_set_path, self._verbose
-            )
+            dependent_on_api_module = self._get_api_module_dependent_items()
+            self._pkgs = list(set(self._pkgs + dependent_on_api_module))
+
+        if json_file_path:
+            if os.path.isdir(json_file_path):
+                json_file_path = os.path.join(json_file_path, "lint_outputs.json")
+        self.json_file_path = json_file_path
+        self.linters_error_list: list = []
+
+    def _get_api_module_dependent_items(self) -> list:
+        changed_api_modules = {
+            pkg.name for pkg in self._pkgs if API_MODULES_PACK in pkg.parts
+        }
+        if changed_api_modules:
+            dependent_items = []
+            for changed_api_module in changed_api_modules:
+                print(
+                    f"Checking for packages dependent on the modified API module {changed_api_module}..."
+                )
+
+                with Neo4jContentGraphInterface() as graph:
+                    print("Updating graph...")
+                    update_content_graph(graph, use_git=True, dependencies=True)
```

Comment:
I'm okay to wait with merging until we implement automatic detection of the need to run update_content_graph() internally in the interface
@ilaner fyi